feat(react-scheduler): add grouping by date (#2630)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/grouping.md

.embedded-demo({ "path": "scheduler-grouping/basic", "showThemeSelector": true })

> ## Grouping by Date

Reviewer: ```suggestion
## Group by Date
```

> The example below shows how to group appointments by date and then by resource. To group by date, use the `GroupingState`'s `groupByDate` property. You can change this order by clicking "Group by Date First" above the Scheduler.

Reviewer 2: In the **following demo**, you can change this order by clicking "Group by Date First" above the Scheduler.

> The example below shows how to group appointments by date and then by resource. To group by date, use the `GroupingState`'s `groupByDate` property. In the following demo, you can change this order by clicking "Group by Date First" above the Scheduler.

Reviewer: ```suggestion
To group appointments by date, set the `GroupingState`'s `groupByDate` property to `true`.

In the example below, appointments are grouped first by date and then by resource. Click the "Group by Date First" switch to change this order.
```
